data/transactions/logic/eval.go

```diff
@@ -2325,6 +2325,33 @@ func opBytesNeq(cx *EvalContext) error {
 	return opNot(cx)
 }
 
+func opBytesModExp(cx *EvalContext) error {
+	result := new(big.Int)
+	last := len(cx.Stack) - 1 // z
+	prev := last - 1          // y
+	pprev := last - 2         // x
+
+	if len(cx.Stack[last].Bytes) > maxStringSize || len(cx.Stack[prev].Bytes) > maxStringSize || len(cx.Stack[pprev].Bytes) > maxStringSize {
```

Comment:
I'm doing this from phone, but `maxStringSize` is the AVM max of 4096? That's unusual for the bmath functions, they normally have a maximum of 128 bytes. I suppose you want more because RSA uses really large keys?

At any rate, it's impossible to supply inputs that are larger than this, so there's no need to check in the opcode.

Reply:
It's ok with me, assuming we can get the cost function to properly account for long inputs. It seems like `bmodexp` wouldn't be very interesting if it can't handle common RSA key sizes.

I'd like to support bigger inputs on the other `b` opcodes too, if we can adjust cost functions appropriately.  They were first done before we could have the cost depend on size. `b+`, for example, would almost certainly be easy to adjust, `b*` might be more complex than simple length dependence. I'm somewhat worried that `bmodexp` is going to be tricky. Anyway, that should be a separate PR someday. (Do your RSA implementations require operating on RSA keys with any other operations?)

Reply:
If it ends up being fast enough, we _could_ pick a cost that accounts for the worst case, which I suppose would three very long inputs.

Or perhaps only scales based on one of the parameters, but accounts for the worst case with the others. (I think `bmodexp` should be linear with respect to the length of the exponent, since it basically performs one operation per bit.)

Reply:
I just read your discussion on cost from the issue more closely.  The dependence on the square of the lengths is unfortunate because it implies a custom cost function. Currently, all costs are "data directed" which is nice because it means we can create specs automatically - we can generate text that describes the cost from the constants provided in the opcode table.

I suppose we can add a way to provide both a Go function and a textual description directly in the `.go` source code. That is somewhat more fragile from the standpoint of modifications to the way we present the spec, but it doesn't see so bad.  It's probably also necessary if we ever want to support larger inputs to `b*` which I suspect is where this really coming from.